src/main/scala/io/iohk/ethereum/domain/branch/Branch.scala

```diff
@@ -2,17 +2,8 @@ package io.iohk.ethereum.domain.branch
 
 import akka.util.ByteString
 
-import io.iohk.ethereum.domain.Block
+sealed trait Branch
 
-/** An interface to manipulate blockchain branches */
-trait Branch {
+case class BestBranch(tipBlockHash: ByteString, tipBlockNumber: BigInt) extends Branch
 
-  /** Returns a block inside this branch based on its number */
-  def getBlockByNumber(number: BigInt): Option[Block]
-
-  /** Returns a block hash for the block at the given height if any */
-  def getHashByBlockNumber(number: BigInt): Option[ByteString]
-
-  /** Checks if given block hash is in this chain. (i.e. is an ancestor of the tip block) */
-  def isInChain(hash: ByteString): Boolean
-}
+case object EmptyBranch extends Branch
```

Comment:
So we'll have multiple types of branch later on, right? For the purposes of the code you're changing now (branch as produced by `getBestBranch`) you could also name the trait `BestBranch` and rename empty branch to `Genesis`, which resolves to a block number of 1. (Or is it 0?)

Reply:
I think we will have several types yes (well, best branch and "other" branch). I don't really know what to do with the `EmptyBranch` as I return it when even the genesis is not found (normally that can happen only during tests). I wonder if we should not add the genesis as a parameter to the block storage, to ensure that it is always there.